consensus/XDPoS/utils/timeout_pool.go

```diff
@@ -0,0 +1,38 @@
+package utils
+
+import "github.com/XinFinOrg/XDPoSChain/common"
+
+type TimeoutPool struct {
+	hashToTimeout map[common.Hash]*Timeout
+	currentRound  Round
+	certSize      int
+}
+
+func NewTimeoutPool(certSize int) *TimeoutPool {
+	return &TimeoutPool{hashToTimeout: make(map[common.Hash]*Timeout), currentRound: 0, certSize: certSize}
+}
+
+func (pool *TimeoutPool) AddTimeout(timeout *Timeout) *TimeoutCert {
+	if timeout.Round != pool.currentRound {
+		return nil
+	}
+	pool.hashToTimeout[timeout.Hash()] = timeout
+	if len(pool.hashToTimeout) >= pool.certSize {
+		signatures := [][]byte{}
+		for h, t := range pool.hashToTimeout {
+			signatures = append(signatures, t.Signature)
+			delete(pool.hashToTimeout, h)
```

Comment:
maybe simply re-create the whole map after create TimeoutCert. Delete 1 by 1 may encounter some edge cases, casing memory leak